sqlglot/dialects/doris.py

```diff
@@ -22,6 +22,48 @@ def _lag_lead_sql(self, expression: exp.Lag | exp.Lead) -> str:
     )
 
 
+# Accept both DATE_TRUNC(datetime, unit) and DATE_TRUNC(unit, datetime)
+def _build_date_trunc(args: list[exp.Expression]) -> exp.Expression:
+    a0, a1 = seq_get(args, 0), seq_get(args, 1)
+
+    def _is_unit_like(e: exp.Expression | None) -> bool:
+        if e is None:
+            return False
+        # Var units like DAY, WEEK, MONTH, QUARTER, YEAR
+        if isinstance(e, exp.Var):
+            return is_date_unit(e) or e.name.lower() in {
+                "hour",
+                "minute",
+                "second",
+            }
+        # String literal units like 'day'
+        if isinstance(e, exp.Literal) and e.is_string:
+            return e.this.strip("'\"").lower() in {
+                "year",
+                "quarter",
+                "month",
+                "week",
+                "day",
+                "hour",
+                "minute",
+                "second",
+            }
+        return False
+
+    # Determine which argument is the unit
+    unit_first = _is_unit_like(a0) and not _is_unit_like(a1)
+    if unit_first:
+        unit, this = a0, a1
+    else:
+        this, unit = a0, a1
+
+    node = exp.TimestampTrunc(this=this, unit=unit)
+
+    # Preserve original argument order for identity rendering via meta
+    node.meta["unit_first"] = unit_first
+    return node
```

Comment:
Why do we need to preserve the order? Sqlglot doesn't have to guarantee the same order, we can canonicalize these `DATE_TRUNC` calls such that, e.g., the unit always comes second.